types/src/block.rs

```diff
@@ -126,624 +149,160 @@ impl From<Block> for RawBlock {
     }
 }
 
-impl ValidateBasic for Header {
-    fn validate_basic(&self) -> Result<(), ValidationError> {
-        if self.version.block != version::BLOCK_PROTOCOL {
-            bail_validation!(
-                "version block ({}) != block protocol ({})",
-                self.version.block,
-                version::BLOCK_PROTOCOL,
-            )
-        }
-
-        if self.chain_id.as_str().len() > MAX_CHAIN_ID_LEN {
-            bail_validation!(
-                "chain id ({}) len > maximum ({})",
-                self.chain_id,
-                MAX_CHAIN_ID_LEN
-            )
-        }
-
-        if self.height.value() == 0 {
-            bail_validation!("height == 0")
-        }
-
-        if self.height.value() == GENESIS_HEIGHT && self.last_block_id.is_some() {
-            bail_validation!("last_block_id == Some() at height {GENESIS_HEIGHT}");
-        }
-
-        if self.height.value() != GENESIS_HEIGHT && self.last_block_id.is_none() {
-            bail_validation!("last_block_id == None at height {}", self.height)
-        }
-
-        // NOTE: We do not validate `Hash` fields because they are type safe.
-        // In Go implementation the validation passes if their length is 0 or 32.
-        //
-        // NOTE: We do not validate `app_hash` because if can be anything
-
-        Ok(())
+#[cfg(all(target_arch = "wasm32", feature = "wasm-bindgen"))]
+#[wasm_bindgen]
+impl Block {
+    /// Block header
+    #[wasm_bindgen(getter)]
+    pub fn get_header(&self) -> JsHeader {
```

Comment:
If I'm not mistaken, wasm-bindgen suggests getters to be without `get_` prefix

Reply:
get_ was added as not to clash with rust name, I'll add rename on js side